File: 7 kyu/Number of People in the Bus/solution.cpp
#include<vector>

unsigned int number(const std::vector<std::pair<int, int>>& busStops){
  int passengers = 0;
  for(auto i: busStops)  passengers += i.first - i.second;
  return passengers;
}
_____________________________________
#include<vector>
using namespace std;


unsigned int number(const vector<pair<int, int>>& busStops) {
    unsigned int answer = 0;
    for (pair<int, int> b: busStops) {
        answer += b.first;
        answer -= b.second;
    }
    return answer;
}
_____________________________________
#include<vector>
#include <numeric>

unsigned int number(const std::vector<std::pair<int, int>>& busStops){
  return std::accumulate(busStops.begin(), busStops.end(), 0, [](unsigned int sum, auto i) { return sum + i.first - i.second; });
}
_____________________________________
#include <numeric>

unsigned int number(const std::vector<std::pair<int, int>>& busStops) {
    return std::accumulate(begin(busStops), end(busStops), 0u,
        [](auto a, auto b) { return a + b.first - b.second; });
}

File: 7 kyu/Number of People in the Bus/solution.kt
fun people(busStops: Array<Pair<Int, Int>>) = busStops.sumBy { it.first - it.second }
_____________________________________
fun people(busStops: Array<Pair<Int, Int>>): Int {
    return busStops.sumBy { (on, off) -> on - off }
}
_____________________________________
fun people(busStops: Array<Pair<Int, Int>>) : Int {
  return busStops.map{it.first - it.second}.reduce{peopleRemaining, passengers -> 
                                  peopleRemaining + passengers}
}
_____________________________________
fun people(busStops: Array<Pair<Int, Int>>) : Int {
  var commonIn = busStops.sumBy { it.first }
  var commonOut = busStops.sumBy { it.second }

  var commonPeople = commonIn - commonOut
  return commonPeople
}
_____________________________________
fun people(busStops: Array<Pair<Int, Int>>) : Int = with(busStops.unzip()) { first.sum() - second.sum() }
